### ISSUE TITLE
[Accessibility] Status changes are not announced for assistive technologies

### ISSUE THREAD

**Issue by eileencho** [jupyter/notebook#5065]
Description: Status changes, such as whether a code cell has finished running, or whether a file has been saved are not being announced for assistive technologies. There are changes on the page that show updated information, such as indication that a code cell is done running based on the number next to the cell, as well as output if the cell has any. For screen reader users, they will only be aware of this change when they specifically navigate the screen reader’s cursor focus to the output rather than being automatically informed. 

Examples:

- After running a code cell in notebooks, there is no indication that code has been run/still running/ output information/ output an error
- There exists a place in the notebook keeping track of when notebook was last saved, but its update after recently saving is not announced
- After running commands in terminal (the instance that can be launched from Jupyter Launch page), there is no information on output of command
- When closing running notebooks and terminals, there is no indication of how many (if any) are still running

Steps to Reproduce (for first example, other examples are similar):

1. Open a notebook
2. Download and start screen reader (such as [Voiceover](https://www.apple.com/voiceover/info/guide/_1124.html) which is already available on Macs, [NVDA](https://www.nvaccess.org/download/) on Windows, [Orca](https://wiki.gnome.org/Projects/Orca) for Linux)
3. Run some code, for example `print(‘Hello World!’)`
4. Note that screen reader doesn’t announce output after running code
5. To make this more clear, turn screen brightness down to 0% and run the code again

[WCAG 2.0 Guidelines Section 4.1.3](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html) (Also has suggestions for [techniques to use](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html#techniques))
[NYU Full Accessibility Audit](https://docs.google.com/spreadsheets/d/1iCiEExo4VDMOwAYtBTCdh2chyx394-1r6YwzmRILF9E/edit?usp=sharing) done with @VickySteeves
